packages/medusa/src/api/routes/admin/inventory-items/create-inventory-item.ts

```diff
@@ -78,54 +77,106 @@ import { validator } from "../../../../utils/validator"
  */
 
 export default async (req, res) => {
-  const validated = await validator(AdminPostInventoryItemsReq, req.body)
-  const { variant_id, ...input } = validated
+  const { variant_id, ...inventoryItemInput } = req.validatedBody
 
-  const inventoryService: IInventoryService =
-    req.scope.resolve("inventoryService")
+  const featureFlagRouter: FlagRouter = req.scope.resolve("featureFlagRouter")
   const productVariantInventoryService: ProductVariantInventoryService =
     req.scope.resolve("productVariantInventoryService")
-  const productVariantService: ProductVariantService = req.scope.resolve(
-    "productVariantService"
-  )
-
-  let inventoryItems = await productVariantInventoryService.listByVariant(
-    variant_id
-  )
-
-  // TODO: this is a temporary fix to prevent duplicate inventory items since we don't support this functionality yet
-  if (inventoryItems.length) {
-    throw new MedusaError(
-      MedusaError.Types.NOT_ALLOWED,
-      "Inventory Item already exists for this variant"
-    )
-  }
 
-  const manager: EntityManager = req.scope.resolve("manager")
+  const createInventoryItemWorkflow = createInventoryItems(req.scope)
 
-  await manager.transaction(async (transactionManager) => {
-    await createInventoryItemTransaction(
+  if (!featureFlagRouter.isFeatureEnabled(ManyToManyInventoryFeatureFlag.key)) {
+    if (!variant_id) {
+      throw new MedusaError(
+        MedusaError.Types.NOT_ALLOWED,
+        "variant_id is required"
+      )
+    }
+
+    createInventoryItemWorkflow.appendAction(
```

Comment:
suggestion: should this step be part of the workflow and depending if the feature flag is enabled we run it or not?

Reply:
I found this cleaner - in the default case you shouldn't be associating an inventory item with a variant as part of the item's creation. So the "correct" workflow should in my mind not include it. 

To ensure that the REST API remains backward compatible, however, we can add the extra step - I find it easier to understand the behavior when feature flags are applied in the layers closest to the client, and that's why I put it here.

That said, I am fine to move it if you think that is best

Reply:
I am fine with keeping it here for now. My thinking is the following, let say that you are the user. You don't necessarily know that the workflow is being customised somewhere in medusa, so you might customise it as well but you don't have the full workflow on front of you since what you know is where the workflow is being located, in the workflow package. Same if we have a visual builder at runtime, since the step is added dynamically at runtime during a request, until this request is fired you will never know that there is another step that should run conditionally. Of course this is out of scope of this pr and something we will have to think of when drafting the guidelines around workflow customisation. Maybe we should add the step outside of the handler but still in that file and have a skip property which would skip it if the flag is not enabled and the endpoint handler would just call the workflow. What i like with the skip is that in a visual mode we will be able to tell the user when a step will run or not which for now is handled by the step itself in other handlers and in this case it is handled outside of the workflow.

Wdyt?

Cc @carlos